x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -193,5 +192,55 @@ def client
         @client ||= build_client
       end
     end
+
+    class SystemIndicesFetcher
+      include LogStash::Util::Loggable
+
+      SYSTEM_INDICES_API_PATH = "_logstash/pipeline"
+
+      def fetch_config(pipeline_ids, client)
+        path_ids = pipeline_ids.join(",")
+        client.get("#{SYSTEM_INDICES_API_PATH}/#{path_ids}")
+      end
+
+      def format_response(response)
```

Comment:
Do we need to expose this method, or could it be done as part of `fetch_config`?